### PR TITLE
fix #18, add option to get state of contract from file for cmd migrate

### LINKED ISSUE #18: State input for migration
Currently, VeriSmart offers to pass transactions that have triggered the target contract via a CSV file ( `-f, --file`). It then retrieves the state changes for each transaction to build the key/value mapping of the current state. However,  passing the final state that is output by `get-state` is not possible yet. Therefore, I would propose the following:

- Rename the `-f, --file` flag to `-x --transaction-input`
- Add flag `-f --state-file` that takes the key/value mapping stored in the json and migrates without accessing the source ledger

### LINKED ISSUE #28: get-state with flag f: writing big JSON objects to file takes too long
This could originate from [contractHelper.js](https://github.com/informartin/VeriSmart/blob/09485987e547de071a15b8e78cf15022fb06bedd/contractHelper.js#L143-145), where JSON object is redundantly parsed into a string, parsed into an object and then parsed into a string again before written to file.

Removing the redundant parsing functions from the pipeline could lead to performance improvements when writing big objects to file.

### PR DESCRIPTION
**Pull request by thecodingshrimp**
fixing issue #18 for now.

But, because get-state will be updated (see PR #23 for details) we can also leave this PR open until it is adapted to the changes.

### CODE REVIEW

**Comment by thecodingshrimp** [comment]
This can now be merged after #26 is merged and #25 is closed.

File: cli/index-migrate [hunk 1]
```diff
@@ -12,30 +14,35 @@ program
...
-        deployment_tx_hash: program.deployment_hash ? program.deployment_hash : undefined,
-        csv_path: program.file ? program.file : undefined,
-        node: node,
-        fat_db: program.fatDb ? program.fatDb : false,
-        targetFile: program.output ? program.output : undefined
-    }
-).then(() => {
-    process.exit();
-});
+contractHelper.extractContractFromJSONFile(program.state_file)
```
Comment (informartin) [inline cli/index-migrate]: It is kind of unintuitive that `extractContractFromJSONFile` is called even if no contract is extracted.

Wrapping the decision if the state is to be extracted from a JSON into a dedicated function may help here.

File: cli/index-shared-attributes [hunk 2]
```diff
@@ -5,7 +5,7 @@ module.exports = (program) => {
         .option('-s, --source <required>', 'Source RPC')
         .option('-c, --contract <required>', 'Source contract address')
         .option('-d, --deployment_hash [optional]', 'Hash of the deployment transaction')
-        .option('-f, --file [optional]', 'Source file (csv) containing TXs to be replayed')
```
Comment (informartin) [inline cli/index-shared-attributes:8]: This change should also be reflected in the README.

**Review by informartin — CHANGES_REQUESTED** [review]

**Comment by thecodingshrimp** [comment]
Writing a big contract to file is now much much faster as well, fixing #28

**Review by informartin — APPROVED** [review]

**Pull request merged**